overlord/snapstate/snapstate.go

```diff
@@ -0,0 +1,109 @@
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// Package snapstate implements the manager and state aspects responsible for the installation and removal of snaps.
+package snapstate
+
+import (
+	"fmt"
+
+	"github.com/ubuntu-core/snappy/i18n"
+	"github.com/ubuntu-core/snappy/overlord/state"
+	"github.com/ubuntu-core/snappy/snappy"
+)
+
+// Install returns a set of tasks for installing snap.
+// Note that the state must be locked by the caller.
+func Install(s *state.State, snap, channel string, flags snappy.InstallFlags) (*state.TaskSet, error) {
+	t := s.NewTask("install-snap", fmt.Sprintf(i18n.G("Installing %q"), snap))
+	t.Set("state", installState{
+		Name:    snap,
+		Channel: channel,
+		Flags:   flags,
+	})
+
+	return state.NewTaskSet(t), nil
+}
+
+// Update initiates a change updating a snap.
+// Note that the state must be locked by the caller.
+func Update(s *state.State, snap, channel string, flags snappy.InstallFlags) (*state.TaskSet, error) {
+	t := s.NewTask("update-snap", fmt.Sprintf(i18n.G("Updating %q"), snap))
+	t.Set("state", installState{
+		Name:    snap,
+		Channel: channel,
+		Flags:   flags,
+	})
+
+	return state.NewTaskSet(t), nil
+}
+
+// Remove returns a set of tasks for removing snap.
+// Note that the state must be locked by the caller.
+func Remove(s *state.State, snap string, flags snappy.RemoveFlags) (*state.TaskSet, error) {
+	t := s.NewTask("remove-snap", fmt.Sprintf(i18n.G("Removing %q"), snap))
+	t.Set("state", removeState{
+		Name:  snap,
+		Flags: flags,
+	})
+
+	return state.NewTaskSet(t), nil
+}
+
+// Purge returns a set of tasks for purging a snap.
+// Note that the state must be locked by the caller.
+func Purge(s *state.State, snap string, flags snappy.PurgeFlags) (*state.TaskSet, error) {
+	t := s.NewTask("purge-snap", fmt.Sprintf(i18n.G("Purging %q"), snap))
+	t.Set("state", purgeState{
+		Name:  snap,
+		Flags: flags,
+	})
+
+	return state.NewTaskSet(t), nil
+}
+
+// Rollback returns a set of tasks for rolling back a snap.
+// Note that the state must be locked by the caller.
+func Rollback(s *state.State, snap, ver string) (*state.TaskSet, error) {
+	t := s.NewTask("rollback-snap", fmt.Sprintf(i18n.G("Rolling back %q"), snap))
+	t.Set("state", rollbackState{
+		Name:    snap,
+		Version: ver,
+	})
+
+	return state.NewTaskSet(t), nil
+}
+
+// SetActive returns a set of tasks for setting a snap active.
+// Note that the state must be locked by the caller.
+func SetActive(s *state.State, snap string, active bool) (*state.TaskSet, error) {
```

Comment:
s/set activate/activate/ everywhere in the function, assuming you're happy about the change.